resource_type.go

```diff
@@ -76,12 +82,34 @@ func (t ResourceType) validate(raw []byte) (ResourceAttributes, *errors.ScimErro
 			return ResourceAttributes{}, scimErr
 		}
 
-		attributes[extension.Schema.ID] = extensionAttributes
+		schemaAttributes[extension.Schema.ID] = extensionAttributes
+	}
+
+	attributes := make(map[string]interface{}, len(commonAttributes)+len(schemaAttributes))
+
+	for k, v := range schemaAttributes {
+		attributes[k] = v
+	}
+
+	for k, v := range commonAttributes {
+		attributes[k] = v
 	}
 
 	return attributes, nil
 }
 
+func (t ResourceType) validateCommonAttributes(m map[string]interface{}) (map[string]interface{}, *errors.ScimError) {
+	if eID, ok := m["externalId"]; ok {
+		externalID, ok := eID.(string)
+		if !ok {
+			return nil, &errors.ScimErrorInvalidValue
+		}
+
+		return map[string]interface{}{"externalId": externalID}, nil
+	}
+	return nil, nil
```

Comment:
should return `map[string]interface{}{}` instead of `nil`. Because you use `len(commonAttributes)` after.